components/constellation/constellation.rs

```diff
@@ -1188,10 +1177,16 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
         self.mem_profiler_chan.send(mem::ProfilerMsg::Exit);
 
-        // TODO: exit before the root browsing context is initialized?
-        debug!("Removing root browsing context.");
-        let root_browsing_context_id = BrowsingContextId::from(self.root_browsing_context_id);
-        self.close_browsing_context(root_browsing_context_id, ExitPipelineMode::Normal);
+        // Close the top-level browsing contexts
+        let top_level_browsing_context_ids: Vec<TopLevelBrowsingContextId> = self.browsing_contexts.values()
+            .filter(|browsing_context| browsing_context.is_top_level())
+            .map(|browsing_context| browsing_context.top_level_id)
+            .collect();
+        for top_level_browsing_context_id in top_level_browsing_context_ids {
+            debug!("Removing top-level browsing context {}.", top_level_browsing_context_id);
+            let browsing_context_id = BrowsingContextId::from(top_level_browsing_context_id);
```

Comment:
If we know the browsing contexts are top level, can't we just get their `id` rather than the `top_level_id` to avoid this conversion?